File: pygit/blob.py
import os


class Blob:

    def __init__(self, data):
        self.data = data
        self._oid = None

    @property
    def type(self):
        return 'blob'

    @property
    def oid(self):
        return self._oid

    @oid.setter
    def oid(self, value):
        self._oid = value

File: gogit/git/blob.go
package git

type Blob struct {
	oid  []byte
	data []byte
}

func (blob *Blob) Init(data []byte) *Blob {
	blob.data = data
	return blob
}

func (blob Blob) Type() string {
	return "blob"
}

func (blob Blob) Data() []byte {
	return blob.data
}

func (blob Blob) Id() []byte {
	return blob.oid
}

func (blob *Blob) SetId(oid []byte) {
	blob.oid = oid
}
